src/telemetry/telemetry.controller.ts

```diff
@@ -90,6 +93,36 @@ export class TelemetryController {
     }
   }
 
+  @ApiExcludeEndpoint()
+  @Post('peers')
+  async writePeerList(
+    @Body(
+      new ValidationPipe({
+        errorHttpStatusCode: HttpStatus.UNPROCESSABLE_ENTITY,
+        transform: true,
+      }),
+    )
+    { nodeId, peers, timestamp }: WritePeerListDto,
+  ): Promise<void> {
+    const promises = peers.map((peer) => {
+      const upload = {
+        sourceId: nodeId,
+        destinationId: peer,
+        timestamp: timestamp,
+      };
+
+      return this.prisma.peerConnection.upsert({
+        create: upload,
+        update: upload,
+        where: {
+          uq_peer_connection_on_source_dest_time: upload,
+        },
+      });
+    });
+
+    await Promise.all(promises);
```

Comment:
We shouldn't use a `Promise.all` on these payloads since that can lead to starvation for the DB.